src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -303,6 +327,19 @@ protected void shadowUpdated(Map<String, Object> desired, Map<String, Object> re
         }
     }
 
+    private void syncShadowDeploymentStatus(Map<String, Object> reported) {
+        // device does not have anything to report
+        if (lastDeploymentStatus.get() == null) {
+            return;
+        }
```

Comment:
Do we expect this to happen in normal cases? When would this occur? The handling doesn't seem right.

Reply:
This would happen when the device receives its first shadow deployment. In that case, the device doesn't have anything to report and there is no need to sync the reported section.

Reply:
`syncShadowDeploymentStatus` is only called when `reported` is non-empty. Is that the case for the first shadow deployment?

Reply:
Yes, for the first shadow deployment the reported section is empty. This will only trigger when the reported section is not null but the lastDeploymentStatus is null (after a device restart)

Reply:
Can you add a log then? I don't like we just return without any log. It will make debugging very difficult.

Reply:
Added log